packages/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,301 @@
+    box-shadow: 0px 2px 4px ${transparentize(0.85, uiColors.black)};
+  `,
+};
+
+const escapeKeyCode = 27;
+
+interface TooltipProps
+  extends Omit<PopoverProps, 'active' | 'spacing' | 'refEl' | 'usePortal'> {
+  /**
+   * A slot for the element used to trigger the `Tooltip`.
+   */
+  trigger: React.ReactElement | Function;
+
+  /**
+   * Determines if a `hover` or `click` event will trigger the opening of a `Tooltip`.
+   */
+  triggerEvent?: TriggerEvent;
+
+  /**
+   * Controls component and determines the open state of the `Tooltip`
+   *
+   * default: `false`
+   */
+  open?: boolean;
+
+  /**
+   * Callback to change the open state of the `Tooltip`.
+   */
+  setOpen?: (
+    open: boolean,
+  ) => void | React.Dispatch<React.SetStateAction<boolean>>;
+
+  /**
+   * Whether the `Tooltip` will be `light` or `dark`.
+   */
+  variant?: Variant;
+
+  /**
+   * id given to `Tooltip` content.
+   */
+  id?: string;
+
+  /**
+   * Callback to determine whether or not `Tooltip` should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+}
+
+/**
+ * # Tooltip
+ *
+ * Tooltip component
+ *
+ * ```
+<Tooltip
+  align='top'
+  justify='start'
+  trigger={<button>trigger</button>}
+  triggerEvent='hover'
+  variant='light'
+>
+  I am an uncontrolled Tooltip!
+</Tooltip>
+```
+ * @param props.children Content to appear inside of Tooltip.
+ * @param props.open Boolean to describe whether or not Tooltip is open.
+ * @param props.setOpen Callback to change the open state of the Tooltip.
+ * @param props.variant Whether the Tooltip should be `dark` or `light`.
+ * @param props.className Classname applied to Tooltip.
+ * @param props.align Alignment of Tooltip relative to trigger: `top`, `bottom`, `left`, `right`.
+ * @param props.justify Justification of Tooltip relative to trigger: `start`, `middle`, `end`.
+ * @param props.trigger Trigger element can be ReactNode or function.
+ * @param props.triggerEvent Whether the Tooltip should be triggered by a `click` or `hover`.
+ * @param props.id id given to Tooltip content.
+ */
+function Tooltip({
+  open: controlledOpen,
+  setOpen: controlledSetOpen,
+  className,
+  children,
+  trigger,
+  variant = Variant.Light,
+  triggerEvent = TriggerEvent.Hover,
+  align = 'top',
+  justify = 'start',
+  id,
+  shouldClose,
+  ...rest
+}: TooltipProps) {
+  const isControlled = !!controlledSetOpen;
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState(false);
+  const open = isControlled ? controlledOpen : uncontrolledOpen;
+  const setOpen = isControlled ? controlledSetOpen : uncontrolledSetOpen;
+
+  const triggerRef = useRef<HTMLElement>(null);
+  const tooltipRef = useRef<HTMLDivElement>(null);
+
+  const tooltipId =
+    id || `tooltip-${Math.floor(Math.random() * Math.floor(10))}`;
+
+  const triggerEventMap = (triggerType: TriggerEvent, triggerProps?: any) => {
+    if (triggerType === 'hover') {
+      return {
+        onMouseEnter: debounce(() => {
```

Comment:
Why are we debouncing these?

Reply:
So that if a user quickly hovers over a tooltip trigger on accident, they aren't seeing the tooltip. Making sure the hover is intentional